tests/Jellyfin.MediaEncoding.Hls.Tests/Playlist/DynamicHlsPlaylistGeneratorTests.cs

```diff
@@ -0,0 +1,101 @@
+﻿using System;
+using Jellyfin.MediaEncoding.Hls.Playlist;
+using Jellyfin.MediaEncoding.Keyframes;
+using Xunit;
+
+namespace Jellyfin.MediaEncoding.Hls.Tests.Playlist
+{
+    public class DynamicHlsPlaylistGeneratorTests
+    {
+        [Theory]
+        [MemberData(nameof(ComputeSegments_Valid_Success_Data))]
+        public void ComputeSegments_Valid_Success(KeyframeData keyframeData, int desiredSegmentLengthMs, double[] segments)
+        {
+            Assert.Equal(segments, DynamicHlsPlaylistGenerator.ComputeSegments(keyframeData, desiredSegmentLengthMs));
+        }
+
+        [Fact]
+        public void ComputeSegments_InvalidDuration_ThrowsArgumentException()
+        {
+            var keyframeData = new KeyframeData(0, new[] { MsToTicks(10000) });
+            Assert.Throws<ArgumentException>(() => DynamicHlsPlaylistGenerator.ComputeSegments(keyframeData, 6000));
+        }
+
+        [Theory]
+        [MemberData(nameof(ComputeEqualLengthSegments_Valid_Success_Data))]
+        public void ComputeEqualLengthSegments_Valid_Success(int desiredSegmentLengthMs, long totalRuntimeTicks, double[] segments)
+        {
+            Assert.Equal(segments, DynamicHlsPlaylistGenerator.ComputeEqualLengthSegments(desiredSegmentLengthMs, totalRuntimeTicks));
+        }
+
+        [Theory]
+        [InlineData(0, 1000000)]
+        [InlineData(1000, 0)]
+        public void ComputeEqualLengthSegments_Invalid_ThrowsInvalidOperationException(int desiredSegmentLengthMs, long totalRuntimeTicks)
+        {
+            Assert.Throws<InvalidOperationException>(() => DynamicHlsPlaylistGenerator.ComputeEqualLengthSegments(desiredSegmentLengthMs, totalRuntimeTicks));
+        }
+
+        [Theory]
+        [InlineData("testfile.mkv", new string[0], false)]
+        [InlineData("testfile.flv", new[] { ".mp4", ".mkv", ".ts" }, false)]
+        [InlineData("testfile.flv", new[] { ".mp4", ".mkv", ".ts", ".flv" }, true)]
```

Comment:
Are the tests on purpose with and without the dot for the extensions?

Reply:
Sort of I suppose. It should be able to handle either, but it's a mistake on my part.